src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -14,6 +14,8 @@ internal sealed class RegexNode
     {
         /// <summary>empty bit from the node's options to store data on whether a node contains captures</summary>
         internal const RegexOptions HasCapturesFlag = (RegexOptions)(1 << 31);
+        /// <summary>Arbitrary number of repetitions of the same character when we'd prefer to represent that as a repeater of that character rather than a string.</summary>
+        internal const int MultiVsRepeaterLimit = 64;
```

Comment:
How did you come up with this - entirely arbitrarily? 😛

Reply:
> entirely arbitrarily?

Somewhat arbitrary.  In our corpus of regexes, the largest single-char repeater has a length of 23.  I chose something significantly larger than that while still being a reasonable length for a string to create.